src/CBLDocument.cc

```diff
@@ -92,6 +92,16 @@ alloc_slice CBLDocument::getRevisionHistory() const {
 }
 
 
+#pragma mark - Utils:
+
+
+void CBLDocument::checkCollectionMatches(CBLCollection* _cbl_nullable myCol, CBLCollection *colParam) {
+    if (myCol && *myCol != *colParam) {
+        C4Error::raise(LiteCoreDomain, kC4ErrorInvalidParameter, "Use document on a wrong collection");
```

Comment:
Can I suggest re-wording the error message?  Maybe something like: "Attempt to save a document into the wrong collection", or "Cannot save a document from collection %1 into collection %2".

Reply:
I also thought of this but it must be generic of operation. This check happens on save, delete (which is a _special_ save) and purge. This is the case for iOS as well.

Reply:
I wondered if that was the case.  I have a couple suggestions, then:
- Simplest: just reword as "Attempt to apply a document to the wrong collection"
- Better: pass a string naming the operation: "Attempt to %1 a document that belongs to collection %2, to collection %3"

Reply:
.NET is having the most descriptive error message for this.

```
"The collection being used for Save does not match the one on the document being saved (" +
"either it is a different collection or a collection from a different database instance).");
```

Let me adapt that message to cover both Save and Delete.